pkg/controller/template/render.go

```diff
@@ -606,3 +607,68 @@ func urlPort(u string) (interface{}, error) {
 		return "", fmt.Errorf("unknown scheme in %s", u)
 	}
 }
+
+func isOpenShiftManagedDefaultLBType(lbType configv1.PlatformLoadBalancerType) bool {
+	switch lbType {
+	case configv1.LoadBalancerTypeOpenShiftManagedDefault:
+		return true
+	default:
+		return false
+	}
+}
+
+func isOpenShiftManagedDefaultLB(cfg RenderConfig) bool {
+	if cfg.Infra.Status.PlatformStatus != nil {
+		lbType := configv1.LoadBalancerTypeOpenShiftManagedDefault
+		switch cfg.Infra.Status.PlatformStatus.Type {
+		case configv1.BareMetalPlatformType:
+			if cfg.Infra.Status.PlatformStatus.BareMetal != nil {
+				if cfg.Infra.Status.PlatformStatus.BareMetal.LoadBalancer != nil {
```

Comment:
Since the feature is TechPreview for now, the upgrades aren't supported when the feature is enabled, and also upgrading a cluster that use the OpenShift managed LB to a User Managed LB is not in the scope for now.

The switch will have to be done by a controller at some point but we haven't designed it at all for now.